omnibus/cookbooks/omnibus-supermarket/attributes/default.rb

```diff
@@ -359,6 +359,10 @@
 # recommend false; set to true as default for backward compatibility
 default['supermarket']['owners_can_remove_artifacts'] = true
 
+# Setting allowed host for supermarket to avoid arbitrary "Host" header injection
+# Set this value to the domain name of your supermarket website e.g. supermarket.chef.io
+default['supermarket']['allowed_host'] = node['supermarket']['fqdn']
```

Comment:
What does this allow specifically and what's the impact to customers? Will people need to change this?

Reply:
Basically this new configuration will help the customers of supermarket prevent any host header injection attack which is a common CVE now a days. Basically with host header injection it's possible to lead the supermarket users to a completely different domain. To prevent this  we have introduced this attribute where the customer can whitelist the hostname of supermarket in the list of allowed hosts. This way any random domain names passed in as Host header in the request to supermarket will be completely discarded thus avoiding such attacks.

So by default this will be set as the FQDN/hostname of the instance where supermarket will be running. In case the user has a custom domain in which supermarket will be running the user simply needs to set the FQDN in the `/etc/supermarket/supermarket.rb` as the intended domain name of supermarket and thus the `allowed_host` will be set accordingly with the command: `supermarket-ctl reconfigure`. So he doesn't need to explicitly set this value.